code/game/objects/items/devices/scanners.dm

```diff
@@ -298,6 +300,7 @@ REAGENT SCANNER
 /obj/item/device/mass_spectrometer/adv
 	name = "advanced mass spectrometer"
 	icon_state = "adv_spectrometer"
+	item_state = "adv_spectrometer"

```

Comment:
No, if I remember how the system works, each item checks for its own icon_state, then makes the item_state the same thing. Might be worth testing, but I've not had problems with it.